linera-chain/src/chain.rs

```diff
@@ -977,18 +977,9 @@ where
         let hash = block.inner().hash();
         let block = block.inner().inner();
         self.execution_state_hash.set(Some(block.header.state_hash));
-        for txn_messages in &block.body.messages {
-            self.process_outgoing_messages(block.header.height, txn_messages)
-                .await?;
-        }
+        self.process_outgoing_messages(block).await?;
 
-        let recipients = block
-            .body
-            .messages
-            .iter()
-            .flatten()
-            .map(|message| message.destination)
-            .collect::<BTreeSet<_>>();
+        let recipients = block.recipients();
```

Comment:
This will now call `recipients` on the `block` twice:
- once in `process_outgoing_messages`
- second time here

vs previously only once. Not sure how important that is for performance 🤷 probably not that much but I wanted to point it out.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3972/commits/3502199f2be73719649e61f83b1e2a8fd0b5ceac.